src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1401,6 +1401,13 @@ private void SelectMatchingUpdatePackages(ShowUpdatePackageOptions updatePackage
                     packageItem.IsSelected = packagesToSelect.Contains(packageItem.Id, StringComparer.OrdinalIgnoreCase);
                 }
             }
+
+            var firstSelected = _packageList.PackageItems.FirstOrDefault(p => p.IsSelected);
```

Comment:
I think as a perf improvement (since both source/destination are sorted by package ID) rather than re-iterating here, could we:
1. do nothing for the `ShouldUpdateAllPackages` case.
2. simply scroll to the first `packageItem` in the first `foreach` iteration.